language/stdlib/modules/Roles.move

```diff
@@ -381,61 +380,238 @@ module Roles {
-        /// be called with an AssociationRootRole argument, because no instances
-        /// of AssociationRootRole can be accessed by another module (and, of course,
-        /// add_privilege_* functions are not called in a way that violates the
-        /// property in this module.
-        ///
-        // invariant forall addr: address where spec_has_role_id(addr):
-        //     exists<Privilege<AssociationRootRole>>(addr)
-        //     ==> (spec_get_role_id(addr) == SPEC_ASSOCIATION_ROOT_ROLE_ID());
-
+        /// **Informally:** For every address, if the address has an
+        /// AssociationRootRole Privilege, then it has a RoleId resource
+        /// with role_id == ASSOCIATION_ROOT_ROLE_ID. (The remaining Role
+        /// privileges are similar, so this explanation is not repeated.)
+        invariant module forall addr: address where exists<Privilege<AssociationRootRole>>(addr):
+             spec_has_role_id(addr)
+                && (spec_get_role_id(addr) == SPEC_ASSOCIATION_ROOT_ROLE_ID());
+
+        /// **Informally:** For every address, if the address has a RootRole resource,
+        /// and the role_id == ASSOCIATION_ROOT_ROLE_ID, then that address also has
+        /// an AssociationRootRole Privilege (also similar to the following properties,
+        /// so also not repeated).
         invariant module forall addr: address where spec_has_role_id(addr):
             (spec_get_role_id(addr) == SPEC_ASSOCIATION_ROOT_ROLE_ID())
-            ==> exists<Privilege<AssociationRootRole>>(addr);
+                ==> exists<Privilege<AssociationRootRole>>(addr);
     }
     spec module {
-        apply AssociationRootRoleMatchesRoleId to public *<T>, *;
+        /// TODO (dd) "except" is a work-around for the testnet weirdness in
+        /// `grant_treasury_compliance_role`
+        /// BUG? (dd) If you delete "public" in front of "*", the "except" doesn't
+        /// seem to work. I haven't been able to reproduce this in a simple test.
+        apply AssociationRootRoleMatchesRoleId to public *, public *<T> except grant_treasury_compliance_role;
     }
 
-    /// **Informally:** Every address that has the treasury compliance
-    /// role ID also has a treasury compliance privilege.
-    ///
-    /// > TODO (dd): Need to add the converse, but that will have the
-    /// same problem as AssociationRootRoleMatchesRoleId due to prover
-    /// limitation.
     spec schema TreasuryComplianceRoleMatchesRoleId {
+        invariant module forall addr: address where exists<Privilege<TreasuryComplianceRole>>(addr):
+             spec_has_role_id(addr)
+                && (spec_get_role_id(addr) == SPEC_TREASURY_COMPLIANCE_ROLE_ID());
+
         invariant module forall addr: address where spec_has_role_id(addr):
             (spec_get_role_id(addr) == SPEC_TREASURY_COMPLIANCE_ROLE_ID())
-             ==> exists<Privilege<TreasuryComplianceRole>>(addr);
+                ==> exists<Privilege<TreasuryComplianceRole>>(addr);
```

Comment:
Same here. Redundant parentheses are considered a style violation in some disciplines, because they are confusing and lead to that other people "learn" from this.